cylc/flow/network/client.py

```diff
@@ -159,7 +162,13 @@ def _socket_options(self):
         self.poller = zmq.Poller()
         self.poller.register(self.socket, zmq.POLLIN)
 
-    async def async_request(self, command, args=None, timeout=None):
+    async def async_request(
+        self,
+        command,
+        args=None,
+        timeout=None,
+        auth_user=None
```

Comment:
If it's quick, it might be a good idea to generalise this to handle arbitrary metadata in a dictionary.

We could do:

```python
# add the shared header
msg.update(self.header)
# add the request metadata
msg['meta'].update(request_meta)
```

This would make it easier to pass through additional information e.g. the app that issued the command or the GraphQL operation name (could be useful if the request contains multiple mutations).